libcxx/include/typeinfo

```diff
@@ -275,13 +275,14 @@ struct __type_info_implementations {
           __impl;
 };
 
-#    if defined(__arm64__) && __has_cpp_attribute(clang::ptrauth_vtable_pointer)
+#    if __has_cpp_attribute(_Clang::__ptrauth_vtable_pointer__)
```

Comment:
@kovdan01 @ahmedbougacha My understanding of the conversation on the previous version of this diff is that this is what we want. Can you please confirm?

Reply:
With the latest changes you've pushed, this LGTM.

Reply:
Yep, this LGTM, thanks!  As mentioned let's give it another day or so to settle the clang side of `ptrauth_type_info_discriminated_vtable_pointer`;  I'm thinking `ptrauth_type_info_vtable_discrimination`

Reply:
I opened #99726 for the codegen side.

@ldionne let's switch `ptrauth_type_info_discriminated_vtable_pointer` below to  `ptrauth_type_info_vtable_pointer_discrimination`